### PR TITLE
Registry Test SetValueWithEnvironmentVariable disabled

### PR DESCRIPTION
**Pull request by Anipik**
Related to https://github.com/dotnet/corefx/issues/28856

### CODE REVIEW

File: src/Microsoft.Win32.Registry/tests/Registry/Registry_SetValue_str_str_obj.cs [hunk 1]
```diff
@@ -116,6 +116,9 @@ public void SetValueWithMultiString()
         [MemberData(nameof(TestEnvironment))]
         public void SetValueWithEnvironmentVariable(string valueName, string envVariableName, string expectedVariableValue)
         {
+            if (envVariableName == "ProgramFiles" && PlatformDetection.IsArmProcess)
```
Comment (danmosemsft) [inline src/Microsoft.Win32.Registry/tests/Registry/Registry_SetValue_str_str_obj.cs]: Please add a comment with the URL of #28856 so it's clear why this is here.

**Comment by Anipik** [comment]
@dotnet-bot test OSX x64 Debug Build

**Comment by Anipik** [comment]
@dotnet-bot test this please

**Comment by Anipik** [comment]
@dotnet-bot test this please

**Comment by Anipik** [comment]
@dotnet/dnceng there is no failures in CI but these 5 legs 
```
Linux x64 Release Build
NETFX x86 Release Build
OSX x64 Debug Build
Windows x64 Debug Build
Windows x86 Release Build
```
Some of these are timing out but the test results look green

**Comment by mmitche** [comment]
@Anipik We had an outage over the weekend due to result processing.  This should be cleared up soon and these failures should go away.

**Pull request merged**